library/core/src/iosMain/kotlin/com/attafitamim/krop/core/images/NSURLImageSrc.kt

```diff
@@ -0,0 +1,38 @@
+package com.attafitamim.krop.core.images
+
+import androidx.compose.ui.unit.IntSize
+import com.attafitamim.krop.core.utils.getImageSize
+import com.attafitamim.krop.core.utils.toImageBitmap
+import com.attafitamim.krop.core.utils.toNSURL
+import com.attafitamim.krop.core.utils.toUIImage
+import kotlinx.cinterop.ExperimentalForeignApi
+import platform.Foundation.NSURL
+
+class NSURLImageSrc(
```

Comment:
internal, right? and the companion object public

Reply:
@vanniktech The companion object contains helper functions, that take fewer or different arguments than the constructor. They are not a replacement and since this is the core module, I wanted to keep some freedom, to make it easier to implement custom logic. Some helper functions might not be as correct or as optmized for some pictures, in that case, the developer has the freedom to, for example, calculate the size of the picture independently and pass it to the constructor.

After forking the library, I made most functions and construcotrs public, before moving them to the core module. 
Maybe we can use `implementation` instead of `api` to depend on `core` from the `ui` module. This way, you won't see this logic if you don't add `core` to your project explicitly. But if you add it, you opt to using an internal api (We can use annotations too).

What do you think?

Reply:
Then for the time being I'd just leave it public. `ui` should expose the `core` via implementation